Healthy/Classes/Modules/Login/LoginViewController.swift

```diff
@@ -57,6 +64,65 @@ private extension LoginViewController {
     }
 }
 
+// MARK: bind to text fields changes
+
+private extension LoginViewController {
+    func bindTextFieldsChanges() {
+        emailTextField.addTarget(self, action: #selector(textDidChange), for: .editingChanged)
+        passwordTextField.addTarget(self, action: #selector(textDidChange), for: .editingChanged)
+    }
+    
+    @objc func textDidChange(_ sender: UITextField) {
+        guard let text = sender.text else { return }
+        
+        if sender == emailTextField {
+            viewModel.updateEmail(text)
+        } else if sender == passwordTextField {
+            viewModel.updatePassword(text)
+        }
+    }
+}
+
+// MARK: - View model binds
+
+private extension LoginViewController {
+    func bindLoadingIndicator() {
+        viewModel.bindLoadingIndicator { [weak self] state in
+            guard let _ = self else { return }
+            // TODO: Update loading state.
+        }
+    }
+    
+    func bindErrorMessage() {
+        viewModel.bindErrorMessage { [weak self] message in
+            guard let _ = self else { return }
+            // TODO: Show error message.
+        }
+    }
+    
+    func bindButtonState() {
+        viewModel.bindButtonEnabled { [weak self] isEnabled in
+            guard let self = self else { return }
+            self.signInButton.isEnabled = isEnabled
+        }
+    }
+    
+    func bindLoginStatus() {
+        viewModel.bindLoginStatus { [weak self] status in
+            guard let _ = self else { return }
```

Comment:
This is a new feature in swift 5.7 which is more elegant. You can check with the team on the minimum version of swift (Xcode too) for the project, if it's higher than 5.7, you can add this change. 
https://www.swiftbysundell.com/articles/swifts-new-shorthand-optional-unwrapping-syntax/

```suggestion
            guard let self else { return }
```